lib/bulk_insert/worker.rb

```diff
@@ -76,12 +76,18 @@ def after_save(&block)
     def save!
       if pending?
         @before_save_callback.(@set) if @before_save_callback
-        compose_insert_query.tap { |query| @connection.execute(query) if query }
+        result = execute_query
         @after_save_callback.() if @after_save_callback
         @set.clear
       end
 
-      self
+      result || []
```

Comment:
so now if you are a block, bulk_insert returns a collection, but if you are the 'value' variant, you get a worker back right?

Reply:
In the lib/bulk_insert file, `save!` is called after the block or after `worker.add_all(values)`. So you should get back a collection with either way.

Reply:
so we have a liskov violation now right if you aren't a block or value variant

Reply:
You mean because it returns a BulkInsert::Worker if no block or values are passed but returns a collection otherwise?
I will revert the last commit so that it always returns a collection.

Reply:
I see how this could be a problem. I'm going to change things so that the result is stored on the worker in an ivar that is readable. Then we don't have to make any changes to what is returned; the worker will still be returned, but we can fetch the result from the worker.